packages/visualizations/src/components/types.ts

```diff
@@ -33,3 +35,10 @@ export type PaletteScale = {
 };
 
 export type ColorScale = GradientScale | PaletteScale;
+
+// We expect an array with couples of values and ExpressionInput and for the last element a default ExpressionInput
+export function isGroupByForMatchExpression(
+    value: ExpressionInputType[]
+): value is [ExpressionInputType, ExpressionInputType, ExpressionInputType] {
+    return value.length >= 3 && value.length % 2 === 1;
+}
```

Comment:
Will tell that it's 3 values or more with an odd number of values (it's straight maplibre.d.ts without the ExpressionSpecification or array of input cases)
```suggestion
export function isGroupByForMatchExpression(
    value: ExpressionInputType[]
): value is [
    ExpressionInputType,
    ExpressionInputType,
    ExpressionInputType,
    ...ExpressionInputType[],
    ExpressionInputType
] {
    return value.length >= 3 && value.length % 2 === 1;
}
```

Reply:
Thanks ! Made the change: https://github.com/opendatasoft/ods-dataviz-sdk/pull/206/commits/69ca6e33d36510cbf958cc7175ac81cddad02b66.